includes/import-products.php

```diff
@@ -139,10 +142,30 @@ function dw_add_woocommerce_product($course) {
 		return ($startA < $startB) ? -1 : 1;
 
 	});
+
+	dw_format_and_save_dates($product, $startDate, $courseParts);
+
+	update_post_meta($product->get_id(), '_product_attributes', $attributes);
+
+	return $product;
+}
+
+/**
+ * @param WC_Product $product
+ * @param DateTime $date
+ * @param array $courseParts
+ *
+ * @throws Exception
+ */
+function dw_format_and_save_dates(WC_Product $product, DateTime $date, array $courseParts) {
+	wp_set_object_terms($product->get_id(), $date->format('d-m-Y'), 'pa_datum', false);
+	wp_set_object_terms($product->get_id(), $date->format('H:i'), 'pa_tijd', false);
+	wp_set_object_terms($product->get_id(), date_i18n('l d F Y', $date->getTimestamp()), 'pa_pretty_date', false);
+
 	$i = 1;
 	foreach($courseParts as $part) {
-		$startDate = new DateTime($part['slots'][0]['startDate']);
-		$endDate = new DateTime($part['slots'][0]['endDate']);
+		$startDate      = new DateTime($part['slots'][0]['startDate']);
+		$endDate        = new DateTime($part['slots'][0]['endDate']);
```

Comment:
@Dirkhuethorst, should we use `isset` or is `$part['slots'][0]['endDate']` always set?

Reply:
Yup, nullable is false for a a slot enddate

Reply:
Then why do we need the `if($endDate)` check a few lines below?

Reply:
We don't, removed it after your last comment, but i accidentally put it back